test/index.js

```diff
@@ -460,4 +470,81 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('submitEncryptionKey', function () {
+    it('sets encryption key data upon submitPassword', async function () {
+      keyringController.cacheEncryptionKey = true;
+      await keyringController.submitPassword(password);
+
+      expect(keyringController.password).toBe(password);
+      expect(keyringController.encryptionSalt).toBe('SALT');
+    });
+
+    it('unlocks the keyrings with valid information', async function () {
+      keyringController.cacheEncryptionKey = true;
+      const returnValue = await keyringController.encryptor.decryptWithKey();
+      const stub = sinon.stub(keyringController.encryptor, 'decryptWithKey');
+      stub.resolves(Promise.resolve(returnValue));
+
+      keyringController.store.updateState({ vault: MOCK_ENCRYPTION_DATA });
+      await keyringController.submitEncryptionKey(MOCK_ENCRYPTION_KEY);
+
+      expect(keyringController.encryptor.decryptWithKey.calledOnce).toBe(true);
+      expect(keyringController.keyrings).toHaveLength(1);
```

Comment:
This might be a more robust test if we locked the keyring controller first before submitting the encryption key. Otherwise this keyring state might be from the initial keyring added in the `beforeEach`. It's not clear it's a result of entering the key.

Reply:
That would also let you get rid of the sinon stub, as you wouldn't need to verify the encryptor was called. The state would prove the key was used.

Though... the password is still set. Hmm.  Ideally the keyring controller would be constructed per-test, so you could do setup differently here and never enter a password. But that would be a larger refactor. Maybe in the meantime you can set `keyringController` to `undefined`, and we can refactor the tests separately.